editoast/editoast_authz/src/roles.rs

```diff
@@ -0,0 +1,252 @@
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
+
+pub trait BuiltinRoleSet:
+    FromStr + Into<&'static str> + Sized + Clone + std::hash::Hash + std::cmp::Eq + std::fmt::Debug
+{
+    fn implies_iter(&self) -> impl IntoIterator<Item = Self>;
+
+    fn as_str(&self) -> &'static str {
+        self.clone().into()
```

Comment:
This would require introducing a trait bound on `&Self`, so it wouldn't be a supertrait, so it would have to be repeated everywhere we use a `BuiltinRoleSet` generic bound :/

On the other hand, I dont think we ever expect those enums to hold data, so we could use a trait bound on Copy (or just expect the clone to be optimized away). @leovalais wdyt?

Reply:
Using `derive(AsRefStr)` solves the issue.